### PR TITLE
Pick up latest `knative.dev/pkg` (k8s 0.22 libs)

### PR DESCRIPTION
**Pull request by mattmoor**
This change picks up the latest `knative.dev/pkg` which contains some codegen changes necessary to use Tekton with the K8s 0.22 libraries, which we've seen a handful of requests for.

As part of this, I have dropped the patch Tekton was carrying to add the subresource field.  cc @vdemeester 

Once this has landed, I will update triggers and chains.

The minimum kubernetes version is still 1.20, so this should NOT be a change to that: https://github.com/knative/pkg/blob/52e42b760b543424fcabcef884cb41e0cc2ff956/version/version.go#L36

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
The pipelines clients are now compatible with k8s.io/client-go v0.22.x
```

### CODE REVIEW

File: hack/spec-gen/main.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ import (
 
 	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 
-	"github.com/go-openapi/spec"
 	"k8s.io/klog"
 	"k8s.io/kube-openapi/pkg/common"
+	spec "k8s.io/kube-openapi/pkg/validation/spec"
```
Comment (mattmoor) [inline hack/spec-gen/main.go:29]: I'm guessing K8s forked this as I had to make this change to get things to pass locally.

**Comment by mattmoor** [comment]
Hmm, I forgot to remove all of the Go 1.16 `//go:build` nonsense, so let's see if CI was upgraded, or I'm going to have to do a lot of manual editing in `nano` 🤞

**Comment by mattmoor** [comment]
cc @wlynch who was asking for this

**Comment by mattmoor** [comment]
I think I stripped the right build tags 🤞

**Comment by mattmoor** [comment]
weird.  one complains with it and the other complains without it 🙃

**Comment by mattmoor** [comment]
Finally.  It looks like most of the codegen gets the new build tags, but the openapi file explicitly must not 🙃

**Review by imjasonh — COMMENTED** [review]
/lgtm

**Pull request merged**